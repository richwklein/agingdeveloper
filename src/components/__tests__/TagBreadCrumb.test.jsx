import React from "react";
import {cleanup, render} from "@testing-library/react";
import TagBreadcrumb from "../TagBreadCrumb";

describe("TagBreadcrumb", () => {
  afterEach(cleanup);

  test("that a category index is generated correctly.", () => {
    const {container} = render(<TagBreadcrumb isCategory={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that a tag index is generated correctly.", () => {
    const {container} = render(<TagBreadcrumb />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that a category breadcrumb is generated correctly.", () => {
    const {container} = render(<TagBreadcrumb name="Microservice" isCategory={true} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toContainHTML("href=\"/category/\"");
    expect(container.firstChild).toContainHTML("microservice");
  });

  test("that a tag breadcrumb is generated correctly.", () => {
    const {container} = render(<TagBreadcrumb name="Gatsbyjs"/>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toContainHTML("href=\"/tag/\"");
    expect(container.firstChild).toContainHTML("gatsbyjs");
  });
});
