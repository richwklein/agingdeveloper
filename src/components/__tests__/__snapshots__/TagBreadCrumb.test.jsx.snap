// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagBreadcrumb that a category breadcrumb is generated correctly. 1`] = `
<div
  class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1x4plm3-MuiDivider-root"
  role="separator"
>
  <span
    class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
  >
    <nav
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
            data-link="internal"
            href="/category/"
          >
            categories
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
          >
            microservice
          </p>
        </li>
      </ol>
    </nav>
  </span>
</div>
`;

exports[`TagBreadcrumb that a category index is generated correctly. 1`] = `
<div
  class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1x4plm3-MuiDivider-root"
  role="separator"
>
  <span
    class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
    >
      categories
    </p>
  </span>
</div>
`;

exports[`TagBreadcrumb that a tag breadcrumb is generated correctly. 1`] = `
<div
  class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1x4plm3-MuiDivider-root"
  role="separator"
>
  <span
    class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
  >
    <nav
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
            data-link="internal"
            href="/tag/"
          >
            tags
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
          >
            gatsbyjs
          </p>
        </li>
      </ol>
    </nav>
  </span>
</div>
`;

exports[`TagBreadcrumb that a tag breadcrumb is generated correctly. 2`] = `
<div
  class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1x4plm3-MuiDivider-root"
  role="separator"
>
  <span
    class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
  >
    <nav
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
            data-link="internal"
            href="/tag/"
          >
            tags
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
          >
            gatsbyjs
          </p>
        </li>
      </ol>
    </nav>
  </span>
</div>
`;

exports[`TagBreadcrumb that a tag index is generated correctly. 1`] = `
<div
  class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1x4plm3-MuiDivider-root"
  role="separator"
>
  <span
    class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
    >
      tags
    </p>
  </span>
</div>
`;
